playbooks/provision-idm-server/setup-idm-dns.yml

```diff
@@ -1,10 +1,24 @@
 ---
+
+- hosts: idm-server
+  tasks:
+  - name: Set required ip address for forward dns
+    set_fact:
+      dns_records: "{{ dns_records|combine({
+        'view': inventory_hostname.split('.')[1:-1],
+        'forward': { 'ip': openstack.private_v4,
+                      'hostname': inventory_hostname.split('.')[0]  },
+        'reverse': { 'hostname': openstack.private_v4.split('.')[-1],
+                      'target': inventory_hostname + '.'  }
+      }, recursive=True) }}"
+    when:
+    - hosting_infrastructure == 'openstack'
+
 - name: 'Copying dns_records from idm hosts'
   hosts: dns-server
   tasks:
-   - set_fact: 
-       dns_records: "{{ hostvars[groups['idm'][0]]['dns_records'] }}" 
-       
+   - set_fact:
+       dns_records: "{{ hostvars[groups['idm'][0]]['dns_records'] }}"
```

Comment:
This should be `dns_records: "{{ hostvars[groups['idm-server'][0]]['dns_records'] }}"` since the group name was refactored

Reply:
I noticed we also started using the `first` filter in some other tasks (i.e. `hostvars[groups['idm-server' | first]['dns_records']`). Is that something that we want to start following across the board? Or is it maybe just a preference thing?

Reply:
@tylerauerbeck That is probably the more 'ansible-y' way to do it, so that sounds like a good direction to take it.

Reply:
@logandonley thought I got them all, but seems like I missed this one. Will update.

@tylerauerbeck more of a personal preference IMHO - I often use `| first` , but for something like this, I think the `[0]` may be cleaner when it's embedded within multiple `[]`... I can go either way really.